File: typescript/src/fizz-buzz/fizz-buzz.ts
export function fizzBuzz(number: number) {
    if (isMultipleOf15(number))
        return 'fizzbuzz';
    if (isMultipleOf3(number))
        return 'fizz';
    if (isMultipleOf5(number))
        return 'buzz';
    return number.toString(10);
}

const isMultipleOf = (multiple: number) => (number: number) => number % multiple === 0;
const isMultipleOf3 = isMultipleOf(3);
const isMultipleOf5 = isMultipleOf(5);
const isMultipleOf15 = isMultipleOf(15);

File: csharp/Prod/FizzBuzz.cs
using System.Linq;

namespace Kata.Prod
{
    public static class FizzBuzz
    {
        public static string Generate(int number)
        {
            switch (GetDividorsPattern(number, 3, 5))
            {
                case "3, _": return "Fizz";
                case "_, 5": return "Buzz";
                case "3, 5": return "FizzBuzz";
                default : return $"{number}";
            }
        }

        private static string GetDividorsPattern(int number, params int[] dividors)
        {
            return dividors.Select(dividor => GetDividorPattern(number, dividor))
                           .Join(", ");
        }

        private static string GetDividorPattern(int number, int dividor)
        {
            return number % dividor == 0
                ? $"{dividor}"
                : "_";
        }
    }
}